src/pubhub/pub_hub.bal

```diff
@@ -8,13 +9,33 @@ import ballerinax/java.jdbc;
 
 import maryamzi/websub.hub.mysqlstore;
 
+// Inbound auth handler for Pub and Hub services
+http:BasicAuthHandler inboundBasicAuthHandler = new(new auth:InboundBasicAuthProvider());
+
 websub:WebSubHub webSubHub = startHubAndRegisterTopic();
 
-listener http:Listener httpListener = new (config:getAsInt("eclk.pub.port", 8181));
+listener http:Listener httpListener = new (config:getAsInt("eclk.pub.port", 8181), config = {
+    auth: {
+        authHandlers: [inboundBasicAuthHandler]
+    },
+    secureSocket: {
```

Comment:
If the keystore and/or truststore are not specified the logic here can fail right.

How about we move this to a function

```suggestion
    secureSocket: getSecureSocket()
```

where get secure socket will return `()` if neither keystore nor truststore are specified (i.e., config lookup returns ""). If only one of the two is specified, we can only set the relevant one and return. WDYT?

Reply:
No, You cant test without them as you need secure socket config to enable auth providers.
When you return () from the method that you specified, following will result.

```
Secure sockets have not been configured in order to enable auth providers. 
```

Reply:
Got it, didn't know the dependency with auth providers.